presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedSet.java

```diff
@@ -130,9 +134,29 @@ public void testBigintSimpleTypedSet()
         }
     }
 
+    @Test
+    public void testMemoryExceeded()
```

Comment:
I'm sorry I didn't make myself clear. I'm asking a single-element array with a super long string element. Note the `'`s in `ARRAY ['" + VERY_LONG_STRING + "']`. This should address your issue?